chainerx_cc/chainerx/routines/connection.cc

```diff
@@ -372,4 +375,73 @@ Array Linear(const Array& x, const Array& w, const absl::optional<Array>& b, uin
     return out_matrix.Reshape(out_shape);
 }
 
+std::vector<Array> _extract_gates(Array x) {
```

Comment:
Could you make fixes to C++ function names?
In C++ we use `CamelCase` without a prefix of `_`.